fineract-core/src/main/java/org/apache/fineract/infrastructure/core/service/database/DatabaseType.java

```diff
@@ -19,5 +19,16 @@
 package org.apache.fineract.infrastructure.core.service.database;
 
 public enum DatabaseType {
-    MYSQL, POSTGRESQL
+
+    MYSQL, //
+    POSTGRESQL, //
+    ;
+
+    public boolean isMySql() {
```

Comment:
Is there any added value of having these "convenience functions"? I think they hide more than they actually help... what happens if we add support for 10 more databases? We have other examples in the code base where these things quickly went out of hand and people started adding all kind of functionality to enums... whereas these files should be really simple and readable. My 2 cents.
Is a simple `MYSQL.equals(dbType)` so much worse (for me it's more readable... imagine static imports of those enums)?
BTW: we should really use "equals" instead "==".

Reply:
This enum is not new. 
For enums the == comparison is better because 
- enums are singleton classes and it never happens that the same enum is not identical with itself
- implementation of equals for enums is actually ==
- == is null safe
- == shows what it is, does not give the impression as if it could be not t
In my opinion enum is better than static string because
- it can have its own functionality
- type safe and value safe
- singleton and identical
- ability to use as list type
- serialised by name (not by value)